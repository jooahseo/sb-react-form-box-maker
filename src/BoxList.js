import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);
  const addBox = (newBox) => {
    setBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };
  const removeBox = (id) => {
    setBoxes(boxes => boxes.filter(box=> box.id !== id))
  }

  return (
    <div>
      <NewBoxForm addBox={addBox}/>
      <div>
        {boxes.map(({ id, width, height, backgroundColor }) => (
            <Box
                key = {id}
                id = {id} 
                removeBox={removeBox}
                width={width} 
                height={height} 
                backgroundColor={backgroundColor} />
        ))}
      </div>
      
    </div>
  );
};

export default BoxList;
