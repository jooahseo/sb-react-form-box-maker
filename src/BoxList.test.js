import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

it("renders without creash", function(){
    render(<BoxList/>);
});

it("matches snapshot", function(){
    const { asFragment } = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
});

it("should add new box", function(){
    const { queryByText, queryByLabelText, queryByTestId } = render(<BoxList/>);
    const widthInput = queryByLabelText("Width");
    const heightInput = queryByLabelText("Height");
    const bgColorInput = queryByLabelText("Background Color");
    const addBtn = queryByText("Add Box");

    expect(queryByTestId("box")).not.toBeInTheDocument();
    expect(queryByText("X")).not.toBeInTheDocument();

    fireEvent.change(widthInput,{target: {value: "30"}});
    fireEvent.change(heightInput,{target: {value: "30"}});
    fireEvent.change(bgColorInput,{target:{value: "pink"}});

    fireEvent.click(addBtn);
    
    expect(queryByTestId("box")).toBeInTheDocument();
    expect(queryByText("X")).toBeInTheDocument();
});