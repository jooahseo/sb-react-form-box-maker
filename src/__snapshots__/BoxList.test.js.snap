// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        Width
      </label>
      <span>
        (50)
      </span>
      <input
        id="width"
        max="500"
        min="10"
        name="width"
        type="range"
        value="50"
      />
      <label
        for="height"
      >
        Height
      </label>
      <span>
        (50)
      </span>
      <input
        id="height"
        max="500"
        min="10"
        name="height"
        type="range"
        value="50"
      />
      <label
        for="backgroundColor"
      >
        Background Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
